lib/src/component_declaration/component_base.dart

```diff
@@ -275,23 +275,24 @@ typedef PropsModifier(Map props);
 abstract class UiProps
     extends Object with MapViewMixin, PropsMapViewMixin, ReactPropsMixin, UbiquitousDomPropsMixin, CssClassPropsMixin
     implements Map {
-  /// Adds an arbitrary prop key-value pair.
-  void addProp(propKey, value) {
+  /// Adds an arbitrary prop key-value pair if [shouldAdd] is true, otherwise, does nothing.
+  void addProp(propKey, value, [bool shouldAdd = true]) {
+    if (!shouldAdd) return;
+
     props[propKey] = value;
   }
 
-  /// Adds a Map of arbitrary props. [props] may be null.
-  void addProps(Map propMap) {
-    if (propMap == null) {
-      return;
-    }
+  /// Adds a Map of arbitrary props if [shouldAdd] is true, otherwise, does nothing.
+  /// [props] may be null.
```

Comment:
#nit [the first sentence of a doc comment should be its own paragraph.](https://www.dartlang.org/guides/language/effective-dart/documentation#do-make-the-first-sentence-a-standalone-paragraph)

There should be an additional newline above this to make two separate paragraphs to adhere to that convention.